### ISSUE TITLE
[Core][Nightly-test] dataset_ingest_train_infer crashed once 

### ISSUE THREAD

**Issue by scv119** [ray-project/ray#26136]
### What happened + What you expected to happen

Error stack trace:

```
Traceback (most recent call last):
  File "ray_sgd_training.py", line 608, in <module>
    train_func=train_func, config=config, callbacks=callbacks, dataset=datasets
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/trainer.py", line 360, in run
    for intermediate_result in iterator:
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/trainer.py", line 758, in __next__
    next_results = self._run_with_error_handling(self._fetch_next_result)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/trainer.py", line 722, in _run_with_error_handling
    return func()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/trainer.py", line 784, in _fetch_next_result
    results = self._backend_executor.get_next_results()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/_internal/utils.py", line 168, in <lambda>
    return lambda *args, **kwargs: ray.get(actor_method.remote(*args, **kwargs))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 2174, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(RuntimeError): ray::BackendExecutor.get_next_results() (pid=340, ip=10.0.0.36, repr=<ray.train._internal.backend_executor.BackendExecutor object at 0x7f55e12a6d90>)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/train/_internal/backend_executor.py", line 397, in get_next_results
    "Some workers returned results while "
RuntimeError: Some workers returned results while others didn't. Make sure that `train.report()` and `train.save_checkpoint()` are called the same number of times on all workers.
```

https://console.anyscale.com/o/anyscale-internal/projects/prj_2xR6uT6t7jJuu1aCwWMsle/clusters/ses_qr6yBMB8YsnrQ9YbFiuSXLsV?command-history-section=command_history

### Versions / Dependencies

latest ray.

### Reproduction script

run dataset_ingest_train_infer

**Comment by jianoaix** [comment]
Is this failing consistently? I saw the latest run was successful.

**Comment by rkooo567** [comment]
@matthewdeng can you also help us what the error message means here?

**Comment by matthewdeng** [comment]
To be concrete:

1. This is not a regression in Train.
2. This seems to be due to flakiness.
3. This should not be a release blocker.
4. We should follow-up and fix Train to handle this type of failure.

**Comment by rkooo567** [comment]
To be clear, this is the bug right?

> (pid=gcs_server) [2022-06-27 05:50:34,678 E 210 210] (gcs_server) gcs_server.cc:296: Failed to get the resource load: GrpcUnavailable: RPC Error message: Connection timed out; RPC Error details: 

It is unexpected from this test. Can anybody confirm?

**Comment by matthewdeng** [comment]
Yeah that's my understanding of it.

**Comment by HuangLED** [comment]
What is the root cause of this error, and any tip how to get around it?

**Comment by HuangLED** [comment]
Our system constantly fail because of this issue, and we observed that it is easier to trigger when there are between 10 - 19 trainers.  It is more stable when there are either more than 19 or less than 10 trainers in parallel.  

Anything rings a bell?

We have been using ray==1.12.1;

**Comment by matthewdeng** [comment]
@HuangLED could you create a separate issue to track this? The `Some workers returned results while others didn't.` error message is not the most descriptive and hides the root cause, which may be different in your case. There have been some improvements in 2.x to improve this.

**Comment by HuangLED** [comment]
> @HuangLED could you create a separate issue to track this? The `Some workers returned results while others didn't.` error message is not the most descriptive and hides the root cause, which may be different in your case. There have been some improvements in 2.x to improve this.

Thanks Matthew. Created a new issue: https://github.com/ray-project/ray/issues/30545

I understand ray is moving forward to 2.x versions. Though given we are relying on RayTrain interface and the whole thing changed since 2.0.  Thus at this point it would be very helpful if there are tips on how to get around it, or if there is any particular 1.x version would mitigate this issue.
